prov/sockets/src/sock_progress.c

```diff
@@ -1166,8 +1166,11 @@ static int sock_pe_progress_buffered_rx(struct sock_rx_ctx *rx_ctx)
 		pe_entry.data_len = 0;
 		pe_entry.buf = 0L;
 		for (i = 0; i < rx_posted->rx_op.dest_iov_len && rem > 0; i++) {
-			if (used_len >= rx_posted->rx_op.dest_iov_len) {
-				used_len -= rx_posted->rx_op.dest_iov_len;
+			/* Try to find an appropriately sized iovec in the
+			 * posted array of iovecs. In the common case, there
+			 * is only one iovec, i.e. a single buffer */
```

Comment:
This comment is misleading.  The looping over the iovec is to find the first iovec entry where the data has not already been consumed.